### PR TITLE
[Az.RecoveryServices.Backup] Added support for customRG with create/modify policy

### PR DESCRIPTION
**Pull request by hiaga**
## Description
* Added support for custom RG with suffix while creating or modifying policy for workload type AzureVM.
* Added TLaD warning https://aka.ms/TLaD.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesBackupProtectionPolicy.cs [hunk 1]
```diff
@@ -106,6 +106,18 @@ public class NewAzureRmRecoveryServicesBackupProtectionPolicy : RSBackupVaultCmd
         [ValidateSet("Days", "Months")]
         public string TierAfterDurationType { get; set; }
 
+        /// <summary>
+        /// Custom resource group name to store the instant recovery points of managed virtual machines.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = ParamHelpMsgs.Policy.AzureBackupResourceGroup)]        
+        public string SnapshotResourceGroup { get; set; }
```
Comment (BethanyZhou) [inline src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesBackupProtectionPolicy.cs:113]: Please update corresponding help MD files

**Review by BethanyZhou — CHANGES_REQUESTED** [review]

File: src/RecoveryServices/RecoveryServices.Backup.Management.Sdk/Generated/Models/IaasVMRecoveryPoint.cs [hunk 2]
```diff
@@ -188,6 +188,5 @@ public IaasVMRecoveryPoint(string recoveryPointType = default(string), System.Da
         /// </summary>
         [JsonProperty(PropertyName = "recoveryPointProperties")]
         public RecoveryPointProperties RecoveryPointProperties { get; set; }
-
```
Comment (BethanyZhou) [inline src/RecoveryServices/RecoveryServices.Backup.Management.Sdk/Generated/Models/IaasVMRecoveryPoint.cs:191]: Please revert this change as generated sdk is not allowed to change manually.

Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Management.Sdk/Generated/Models/IaasVMRecoveryPoint.cs]: reverted

File: src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesBackupProtectionPolicy.cs (hunk 1 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/NewAzureRmRecoveryServicesBackupProtectionPolicy.cs]: Updated

**Review by BethanyZhou — DISMISSED** [review]

File: src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md [hunk 3]
```diff
@@ -55,14 +57,14 @@ $RetPol.WeeklySchedule.DurationCountInWeeks = 365
 $vault = Get-AzRecoveryServicesVault -ResourceGroupName "azurefiles" -Name "azurefilesvault"
 $Pol= Get-AzRecoveryServicesBackupProtectionPolicy -Name "TestPolicy" -VaultId $vault.ID
 $Pol.SnapshotRetentionInDays=5
-Set-AzRecoveryServicesBackupProtectionPolicy -Policy $Pol -SchedulePolicy $SchPol -RetentionPolicy $RetPol
+Set-AzRecoveryServicesBackupProtectionPolicy -Policy $Pol -SchedulePolicy $SchPol -RetentionPolicy $RetPol -AzureBackupSnapshotRGName "snapshotResourceGroupPrefix" -AzureBackupSnapshotRGName "snapshotResourceGroupSuffix"
```
Comment (BethanyZhou) [inline src/RecoveryServices/RecoveryServices/help/Set-AzRecoveryServicesBackupProtectionPolicy.md:60]: "Az.RecoveryServices","Set-AzRecoveryServicesBackupProtectionPolicy","1","19","Invalid_Parameter_Name","5011","1","Set-AzRecoveryServicesBackupProtectionPolicy -AzureBackupSnapshotRGName is not a valid parameter name.","-AzureBackupSnapshotRGName","Check validity of the parameter -AzureBackupSnapshotRGName."

**Comment by BethanyZhou** [comment]
Hi @hiaga, verified the example issues on my local environment. Issues is gone, seems like this is a tool bug. Please suppress these  issue to pass ci verification.

**Review by BethanyZhou — APPROVED** [review]
